lib/spack/spack/cmd/buildcache.py

```diff
@@ -53,7 +53,7 @@ def setup_parser(subparser):
                         type=str, default=None,
                         help="Key for signing.")
     create.add_argument('-d', '--directory', metavar='directory',
-                        type=str, default='.',
+                        type=str, default=None,
```

Comment:
The help should be modified to state that name of mirrors are also accepted. As an aside `--directory name-of-mirror` feels a bit weird .

Reply:
Turns out mirror names don't work as you would expect. I didn't try with the url prefix though.

Reply:
:sob:

Reply:
> Turns out mirror names don't work as you would expect.

How so?  I do not observe an issue with using mirror names.

```
 $ spake install --fake zlib
...
 $ spack mirror list
==> No mirrors configured.
# "local" is the path to a directory on the filesystem (i.e.: "./local")
 $ spack buildcache create -d local --unsigned zlib
...
 $ find local/
local/
local//build_cache
local//build_cache/index.html
local//build_cache/darwin-mojave-x86_64-clang-10.0.0-apple-zlib-1.2.11-uw6agnye4za7cpgyyaaejrzorhbhqno5.spec.yaml
local//build_cache/darwin-mojave-x86_64
local//build_cache/darwin-mojave-x86_64/clang-10.0.0-apple
local//build_cache/darwin-mojave-x86_64/clang-10.0.0-apple/zlib-1.2.11
local//build_cache/darwin-mojave-x86_64/clang-10.0.0-apple/zlib-1.2.11/darwin-mojave-x86_64-clang-10.0.0-apple-zlib-1.2.11-uw6agnye4za7cpgyyaaejrzorhbhqno5.spack
```
```
 $ spack mirror add local ./other-dir
# "local" is now the name of a mirror
 $ spack buildcache create -d local --unsigned zlib
...
 $ find other-dir/
other-dir/
other-dir//build_cache
other-dir//build_cache/index.html
other-dir//build_cache/darwin-mojave-x86_64-clang-10.0.0-apple-zlib-1.2.11-uw6agnye4za7cpgyyaaejrzorhbhqno5.spec.yaml
other-dir//build_cache/darwin-mojave-x86_64
other-dir//build_cache/darwin-mojave-x86_64/clang-10.0.0-apple
other-dir//build_cache/darwin-mojave-x86_64/clang-10.0.0-apple/zlib-1.2.11
other-dir//build_cache/darwin-mojave-x86_64/clang-10.0.0-apple/zlib-1.2.11/darwin-mojave-x86_64-clang-10.0.0-apple-zlib-1.2.11-uw6agnye4za7cpgyyaaejrzorhbhqno5.spack
```

Reply:
Maybe because I put quotes around the name (see the third command). Using a url or a name without quotes works as you designed.
```
(flake8) [gartung@gartung spack]$ spack buildcache create -f -d file:///build/test /ex3qnyb  
==> Buildcache files will be output to file:///build/test/build_cache
==> creating binary cache file for package patchelf@0.10%gcc@4.8.5 arch=linux-scientific7-x86_64 
==> Buildache for "patchelf@0.10%gcc@4.8.5 arch=linux-scientific7-x86_64" written to 
 file:///build/test/build_cache/linux-scientific7-x86_64/gcc-4.8.5/patchelf-0.10/linux-scientific7-x86_64-gcc-4.8.5-patchelf-0.10-ex3qnybcwz2icdzfjlawuju7y63zslu3.spack
(flake8) [gartung@gartung spack]$ spack buildcache create -f -d local /ex3qnyb  
==> Buildcache files will be output to file:///build/test/spack/local/build_cache
==> creating binary cache file for package patchelf@0.10%gcc@4.8.5 arch=linux-scientific7-x86_64 
==> Buildache for "patchelf@0.10%gcc@4.8.5 arch=linux-scientific7-x86_64" written to 
 file:///build/test/spack/local/build_cache/linux-scientific7-x86_64/gcc-4.8.5/patchelf-0.10/linux-scientific7-x86_64-gcc-4.8.5-patchelf-0.10-ex3qnybcwz2icdzfjlawuju7y63zslu3.spack
(flake8) [gartung@gartung spack]$ spack buildcache create -f -d "local" /ex3qnyb  
==> Buildcache files will be output to file:///build/test/spack/local/build_cache
==> creating binary cache file for package patchelf@0.10%gcc@4.8.5 arch=linux-scientific7-x86_64 
==> Buildache for "patchelf@0.10%gcc@4.8.5 arch=linux-scientific7-x86_64" written to 
 file:///build/test/spack/local/build_cache/linux-scientific7-x86_64/gcc-4.8.5/patchelf-0.10/linux-scientific7-x86_64-gcc-4.8.5-patchelf-0.10-ex3qnybcwz2icdzfjlawuju7y63zslu3.spack
```

Reply:
```
...
(flake8) [gartung@gartung spack]$ spack buildcache create -f -d local /ex3qnyb  
==> Buildcache files will be output to file:///build/test/spack/local/build_cache
,,,
(flake8) [gartung@gartung spack]$ spack buildcache create -f -d "local" /ex3qnyb  
==> Buildcache files will be output to file:///build/test/spack/local/build_cache
...
```

Am I missing something?  The output is exactly the same in both cases.  This makes sense, because the quotes are consumed by the shell's parser before invoking the Spack command. To Spack, the args received from `buildcache create -d local` and `buildcache create -d "local"` should be identical.

Reply:
I had a mirror named "test" not "local". Sorry for the incorrect assumptions.